### PR TITLE
Added mysql finder and removed the c++11 warning for obfuscate_pass

### PR DESCRIPTION
**Pull request by WD-stefaang**
I'm working on Ubuntu and wanted to use the libmysqlclient-dev package.

### CODE REVIEW

File: URIResolver/CMakeLists.txt [hunk 1]
```diff
@@ -9,12 +9,15 @@ find_package(Boost REQUIRED)
...
 
 link_directories(${USD_LIBRARY_DIR})
 
-find_library(MYSQL_CONNECTOR_LIBRARY
-             name libmysqlclient.a
-             HINTS ${MYSQL_CONNECTOR_ROOT}/lib)
+find_library(MYSQL_CLIENT_LIBRARY
+		NAMES "libmysqlclient.a"
+		PATHS ${MYSQL_CONNECTOR_ROOT}/lib
+			/usr/lib/x86_64-linux-gnu
```
Comment (sirpalee) [inline URIResolver/CMakeLists.txt]: Why do we need to call find_library and find_package at the same time? If there is a find module for MySql, let's use the library found in that module.

File: cmake/FindMySQL.cmake [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+			  "$ENV{${MYENV}}/MySQL/*/lib"
+			  "$ENV{SYSTEMDRIVE}/MySQL/*/lib" )
+else()
+	find_path( MYSQL_INCLUDE_DIR
+		NAMES "mysql.h"
+		PATHS "/usr/include/mysql"
+			  "/usr/local/include/mysql"
+			  "/usr/mysql/include/mysql" )
+	
+	find_library( MYSQL_LIBRARY
```
Comment (sirpalee) [inline cmake/FindMySQL.cmake]: What's the way to define a custom location for the mysql lib?

File: cmake/FindMySQL.cmake [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+		NAMES "mysqlclient" "mysqlclient_r"
```
Comment (sirpalee) [inline cmake/FindMySQL.cmake]: This means the dynamic library will always be preferred over the static if both exist in the same directory. We need a way to control (see how boost handles choosing between static and dynamic libraries).

File: cmake/FindMySQL.cmake [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+			  "$ENV{${MYENV}}/MySQL/*/include"
+			  "$ENV{SYSTEMDRIVE}/MySQL/*/include" )
+	
+	find_library( MYSQL_LIBRARY
+		NAMES "mysqlclient" "mysqlclient_r" "libmysql"
+		PATHS "$ENV{PROGRAMFILES}/MySQL/*/lib"
+			  "$ENV{${MYENV}}/MySQL/*/lib"
+			  "$ENV{SYSTEMDRIVE}/MySQL/*/lib" )
+else()
+	find_path( MYSQL_INCLUDE_DIR
```
Comment (sirpalee) [inline cmake/FindMySQL.cmake]: What's the preferred mechanism to define a custom location for headers?

File: cmake/FindMySQL.cmake [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  IF (MySQL_FIND_REQUIRED)
+    MESSAGE(STATUS "Looked for MySQL libraries named ${MYSQL_NAMES}.")
+    MESSAGE(FATAL_ERROR "Could NOT find MySQL library")
+  ENDIF (MySQL_FIND_REQUIRED)
+ENDIF (MYSQL_FOUND)
+
+MARK_AS_ADVANCED(
+  MYSQL_LIBRARY
+  MYSQL_INCLUDE_DIR
+)
```
Comment (sirpalee) [inline cmake/FindMySQL.cmake:85]: Can we use find_package_handle_standard_args here?

Also, looking at mysql_version.h it should be trivial to parse the version number for the mysql package, so in the future we can specify which version we require. Look at FindUSD.cmake as an example.

**Review by sirpalee — CHANGES_REQUESTED** [review]

File: URIResolver/CMakeLists.txt (hunk 1 above)
Comment (WD-stefaang) [reply URIResolver/CMakeLists.txt]: Sure, will fix.

File: cmake/FindMySQL.cmake (hunk 5 above)
Comment (WD-stefaang) [reply cmake/FindMySQL.cmake:85]: Done.

**Comment by WD-stefaang** [comment]
I've tested dynamic library, static library and custom library path on my Ubuntu machine. 
https://gist.github.com/WD-stefaang/e8c5c4b46e13bf3bf80f88b62ca406c6

Use MYSQL_ROOT environment var to set a custom installation root.
Use MYSQL_USE_STATIC_LIB cmake option to use the static library.

**Comment by sirpalee** [comment]
Thanks for the updates! I'll pull the changes locally soon and see how it behaves with our internal build system.

**Comment by WD-stefaang** [comment]
I had just copied the cmake from another project, it's the first time I wrote cmake code myself, so feel free to add comments.. e.g. I'm not sure if the mark_as_advanced calls are necessary.

**Comment by sirpalee** [comment]
Hey! I merged in your changes and made some minimal modifications to match cmake to our more current internal standards. Can you give the latest master a go and let me know if everything is okay on your end?

I found one significant issue, while CMAKE_CXX_STANDARD is the way to go, it's only available since 3.1. So I needed to bump up cmake_minimum_required to 3.1.

**Comment by WD-stefaang** [comment]
Thanks. 
I see the readme.md still refers to cmake 2.8+ and the mysql find script docstring doesn't include the new options.

**Pull request merged**